google/cloud/aiplatform/training_jobs.py

```diff
@@ -281,46 +333,474 @@ def package_and_copy_to_gcs(
 
+        Args:
+            dataset (aiplatform.Dataset):
+                AI Platform to fit this training against. Custom training script should
+                retrieve datasets through passed in environement variables uris:
+
+                os.environ["AIP_TRAINING_DATA_URI"]
+                os.environ["AIP_VALIDATION_DATA_URI"]
+                os.environ["AIP_TEST_DATA_URI"]
+
+                Additionally the dataset format is passed in as:
+
+                os.environ["AIP_DATA_FORMAT"]
+            model_display_name (str):
+                If the script produces a managed AI Platform Model. The display name of
+                the Model. The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            base_output_dir (str):
+                GCS output directory of job. If not provided a
+                timestamped directory in the staging directory will be used.
+            args (Dict[str, str]):
+                Key-value pair or command line arguments to be passed to the Python
+                script.
+            replica_count (int):
+                The number of worker replicas.
+            accelerator_type (str):
+                Hardware accelerator type. One of ACCELERATOR_TYPE_UNSPECIFIED,
+                NVIDIA_TESLA_K80, NVIDIA_TESLA_P100, NVIDIA_TESLA_V100, NVIDIA_TESLA_P4,
+                NVIDIA_TESLA_T4, TPU_V2, TPU_V3
+            accelerator_count (int):
+                The number of accelerators to attach to a worker replica.
+            training_fraction_split (float):
+                The fraction of the input data that is to be
+                used to train the Model.
+            validation_fraction_split (float):
+                The fraction of the input data that is to be
+                used to validate the Model.
+            test_fraction_split (float):
+                The fraction of the input data that is to be
+                used to evaluate the Model.
 
-class CustomTrainingJob(TrainingJob):
-    pass
+        Returns:
+            model: The trained AI Platform Model resource or None if training did not
+                produce an AI Platform Model.
+
+        Raises:
+            RuntimeError if Training job has already been run, staging_bucket has not
+                been set, or model_display_name was provided but required arguments
+                were not provided in constructor.
+            NotImplementedError more then one replica.
+            ValueError if accelerator type is not valid.
+        """
+
+        def format_args(args: Dict[str, str]) -> List[str]:
+            """Formats key-value arguments.
+
+            Args:
+                args (Dict[str, str]):
+                    Required: key-value pair or arguments.
+            Returns:
+                arg_string: List of formated arguments.
+            """
+            return [f"--{key}={value}" for key, value in args.items()]
```

Comment:
In particular, the user-provided package should not accept positional arguments?

Reply:
Acked, resolved, and explained above.